src/Microsoft.AspNetCore.Razor.Language/Legacy/MarkupParser.cs

```diff
@@ -0,0 +1,916 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language.Syntax.InternalSyntax;
+
+namespace Microsoft.AspNetCore.Razor.Language.Legacy
+{
+    internal partial class HtmlMarkupParser : TokenizerBackedParser<HtmlTokenizer>
+    {
+        private const string ScriptTagName = "script";
+
+        private static readonly char[] ValidAfterTypeAttributeNameCharacters = { ' ', '\t', '\r', '\n', '\f', '=' };
+        private static readonly SyntaxToken[] nonAllowedHtmlCommentEnding = new[]
+        {
+            SyntaxFactory.Token(SyntaxKind.Text, "-"),
+            SyntaxFactory.Token(SyntaxKind.Bang, "!"),
+            SyntaxFactory.Token(SyntaxKind.OpenAngle, "<"),
+        };
+
+        //From http://dev.w3.org/html5/spec/Overview.html#elements-0
+        private readonly ISet<string> _voidElements = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "area",
+            "base",
+            "br",
+            "col",
+            "command",
+            "embed",
+            "hr",
+            "img",
+            "input",
+            "keygen",
+            "link",
+            "meta",
+            "param",
+            "source",
+            "track",
+            "wbr"
+        };
+
+        public HtmlMarkupParser(ParserContext context)
```

Comment:
I'm not reviewing this stuff in detail, I'm just assuming that it was moved to another file unchanged.

Reply:
LMK if that's wrong

Reply:
I wouldn't skip them completely. They were mostly moved but I refactored how leading or trailing whitespace is handled etc. I think it's worth a skim atleast

Reply:
OK I think it is wrong. Looking at this in detail then.